pkg/odo/cli/service/create.go

```diff
@@ -45,46 +49,96 @@ func NewServiceCreateOptions() *ServiceCreateOptions {
-	o.serviceType = args[0]
-	// if only one arg is given, then it is considered as service name and service type both
-	o.serviceName = o.serviceType
-	// if two args are given, first is service type and second one is service name
-	if len(args) == 2 {
-		o.serviceName = args[1]
+	if len(args) == 0 || !cmd.HasFlags() {
+		o.interactive = true
 	}
+
 	o.Context = genericclioptions.NewContextCreatingAppIfNeeded(cmd)
-	return err
+
+	client := o.Client
+
+	var class scv1beta1.ClusterServiceClass
+	if o.interactive {
+		classesByCategory, _ := client.GetServiceClassesByCategory()
+		class, o.serviceType = ui.SelectClassInteractively(classesByCategory)
+
+		plans, _ := client.GetMatchingPlans(class)
+
+		var svcPlan scv1beta1.ClusterServicePlan
+		// if there is only one available plan, we select it
+		if len(plans) == 1 {
+			for k, v := range plans {
+				o.plan = k
+				svcPlan = v
+			}
+			glog.V(4).Infof("Plan %s was automatically selected since it's the only one available for service %s", o.plan, o.serviceType)
+		} else {
+			// otherwise select the plan interactively
+			o.plan = ui.SelectPlanNameInteractively(plans, "Which service plan should we use ")
+			svcPlan = plans[o.plan]
+		}
+
+		o.parametersMap = ui.EnterServicePropertiesInteractively(svcPlan)
+		o.serviceName = ui.EnterServiceNameInteractively(o.serviceType, "How should we name your service ", validateNameFn)
+	} else {
+		o.serviceType = args[0]
+		// if only one arg is given, then it is considered as service name and service type both
+		o.serviceName = o.serviceType
+		// if two args are given, first is service type and second one is service name
+		if len(args) == 2 {
+			o.serviceName = args[1]
+		}
+	}
+
+	return
+}
+
+var validateNameFn = func(i interface{}) error {
+	s := i.(string)
+	return util.ValidateName(s)
 }
 
 // Validate validates the ServiceCreateOptions based on completed values
 func (o *ServiceCreateOptions) Validate() (err error) {
+	// if we are in interactive mode, all values are already valid
+	if o.interactive {
+		return nil
+	}
+
 	// make sure the service type exists
-	matchingService, err := svc.GetSvcByType(o.Client, o.serviceType)
+	classPtr, err := o.Client.GetClusterServiceClass(o.serviceType)
 	if err != nil {
 		return fmt.Errorf("unable to create service because Service Catalog is not enabled in your cluster:\n%v", err)
 	}
-	if matchingService == nil {
+	if classPtr == nil {
 		return fmt.Errorf("service %v doesn't exist\nRun 'odo service catalog' to see a list of supported services.\n", o.serviceType)
 	}
 	if len(o.plan) == 0 {
+		plans, err := o.Client.GetMatchingPlans(*classPtr)
+		if err != nil {
+			return err
+		}
 		// when the plan has not been supplied, if there is only one available plan, we select it
-		if len(matchingService.PlanList) == 1 {
-			o.plan = matchingService.PlanList[0]
+		if len(plans) == 1 {
+			for k := range plans {
```

Comment:
Why are we looping since the length is only 1?

Reply:
How else do you retrieve the unique key in a map?

Reply:
Oops! didn't notice it was map. Sorry, my bad :)